### DISCUSSION TITLE
How to generate  Postman collections and tests only for the tests/paths specified in Portman configuration file?

### DISCUSSION THREAD

**Discussion by armen-ch** [apideck-libraries/portman#470] [Q&A]
Hello,

I'm trying to run variation tests generated with Portman using "_-runNewman true_" from the OpenAPI spec file. I have a Portman config file that only has a **VariationTests** property block that specifies overwrite and variation tests. And I want Portman to only generate these variation tests to allow Newman to run only these tests. But it generates and runs a Postman collection for all paths/operations that exist in the OpenAPI spec file. How can I configure Portman (or maybe Newman) to **generate/run only certain tests** and not the entire collection from the OpenAPI spec?

Thanks

**Comment by thim81** [comment]
Hi,

Portman comes packaged with [openapi-format](https://github.com/thim81/openapi-format#openapi-filter-options), so you can use  the `--filterFile` parameter to pass a path to openapi-format config file (oas-format-filter.json)

We have provided an example setup: https://github.com/apideck-libraries/portman/tree/main/examples/cli-filtering

**Reply by armen-ch** [reply]
Thank you @thim81 for the response. This filtering looks great and would be quite helpful feature. But looks like it will not resolve my problem. I'll try to clarify my case.
For instance, I have a oas yaml file which specifies multiple paths for microservice. Something like the one in the screenshot:
![Screenshot-oas-yml](https://user-images.githubusercontent.com/70989770/225387040-b8d52bf7-f662-48e2-8bbd-f316c3c63eb1.png)

Now I want to generate only one test for 401 response and therefore I specify one variation test for that path in Portman config file like in the below screenshot:
![Screenshot-portman-config](https://user-images.githubusercontent.com/70989770/225387565-f6214267-c923-488f-ba5e-b01497453a51.png)

After running Portman I expect that I'll have the generated file which contains only /v1/products endpoint call with bogus_token and test which verifies 401 response status code. 
But actually it generates request collection for the entire api spec + that one variation test.
If I'm passing this generated file to Newman, it calls all the endpoints (which actually I don't want to happen) + executes that one variation test. But actually want to execute that variation test only. 
To achieve this, I should either be able to prevent the creation of the Postman collection for all OpenAPI spec endpoints and be able to generate tests specified only in the Portman config file explicitly or be able to prevent them(collection endpoints) from being called/executed in Newman. 
Please advise. Or let me know if you need further clarification.

thanks

**Reply by thim81** [reply]
Would it be helpful for you to use the `--folder [folderName]` Newman CLI option?
It allows you to specify to run only a certain folder.
https://learning.postman.com/docs/collections/using-newman-cli/newman-options/#setup

**Reply by armen-ch** [reply]
It works for me. Don't you think it worth to add test type filtering to Portman or may be to oas-format-filter? I think it would be useful.
But anyways, thanks for the help @thim81!

**Comment by armen-ch** [comment]
Resolved by using --folder [folderName] Newman CLI option
